rts/Lua/LuaOpenGL.cpp

```diff
@@ -5661,5 +5666,69 @@ int LuaOpenGL::GetMapRendering(lua_State* L)
 	return 0;
 }
 
+
+/**
+ * @function gl.ObjectLabel labels an object for use with debugging tools
+ * @param objectTypeIdentifier GLenum Specifies the type of object being labeled.
+ * @param objectID GLuint Specifies the name or ID of the object to label.
+ * @param label string A null-terminated string containing the label to be assigned to the object.
+ * @treturn nil
+ */
+int LuaOpenGL::ObjectLabel(lua_State* L) {
+	GLenum identifier = (GLenum)luaL_checkinteger(L, 1);
+	if (identifier != GL_BUFFER && identifier != GL_SHADER && identifier != GL_PROGRAM &&
+		identifier != GL_VERTEX_ARRAY && identifier != GL_QUERY && identifier != GL_PROGRAM_PIPELINE &&
+		identifier != GL_TRANSFORM_FEEDBACK && identifier != GL_TEXTURE && identifier != GL_RENDERBUFFER &&
+		identifier != GL_FRAMEBUFFER) {
+		return 0;
+	}
+    GLuint objectID = (GLuint)luaL_checkinteger(L, 2);
+    const char* label = luaL_checkstring(L, 3);
+
+    // Ensure that identifier is a valid GLenum
+    // The length should be passed as -1 because the strings are null terminated
+    // Check that the GLuint name refers to a valid openGL object
+    // Check that the label is a valid string
```

Comment:
Do these "check this, check that" comments mean that the code is WIP and those checks should be added?